crates/matrix-sdk-ui/tests/integration/timeline/mod.rs

```diff
@@ -67,6 +68,69 @@ mod thread;
 
 pub(crate) mod sliding_sync;
 
+#[async_test]
+async fn test_timeline_is_threaded() {
+    let server = MatrixMockServer::new().await;
+    let client = server.client_builder().build().await;
+
+    let room_id = room_id!("!a98sd12bjh:example.org");
+    let room = server.sync_joined_room(&client, room_id).await;
+
+    server.mock_room_state_encryption().plain().mount().await;
+
+    {
+        // A live timeline isn't threaded.
+        let timeline = room.timeline().await.unwrap();
+        assert!(timeline.is_threaded().not());
+    }
+
+    {
+        // A thread timeline is threaded.
+        let timeline = TimelineBuilder::new(&room)
+            .with_focus(TimelineFocus::Thread { root_event_id: owned_event_id!("$thread_root") })
+            .build()
+            .await
+            .unwrap();
+        assert!(timeline.is_threaded());
+    }
+
+    {
+        let f = EventFactory::new();
+        let event = f
+            .text_msg("hello world")
+            .event_id(event_id!("$target"))
+            .room(room_id)
+            .sender(&ALICE)
+            .into_event();
+        server.mock_room_event_context().ok(event, "", "", Vec::new()).mount().await;
+
+        // An event-focused timeline isn't threaded.
```

Comment:
Yet?

Reply:
Indeed! I don't think it's worth commenting "yet" in the code, though, since this is going to happen, and it's unlikely that we'll remember to update it in the test :)